warehouse/query-core/src/main/java/org/apache/commons/jexl3/parser/JexlNodes.java

```diff
@@ -65,24 +68,14 @@ public static int id(JexlNode n) {
     public static <T extends JexlNode> T newInstanceOfType(T node) {
         try {
             @SuppressWarnings("rawtypes")
-            Constructor constructor = node.getClass().getConstructor(Integer.TYPE);
+            Constructor constructor = node.getClass().getDeclaredConstructor(Integer.TYPE);
```

Comment:
I believe this constructor appears to be public on the nodes.  Can we not still used getConstructor ?

Reply:
There are quite a few nodes which do not have a public constructor matching this signature.